lib/dor/services/client/marcxml.rb

```diff
@@ -21,6 +23,37 @@ def catkey(barcode:)
 
           raise_exception_based_on_response!(resp)
         end
+
+        # Gets MARCXML corresponding to a barcode or catkey
+        # @param barcode [String] required string representing a barcode
+        # @param catkey [String] required string representing a catkey
+        # @raise [NotFoundResponse] when the response is a 500 with "Record not found in Symphony"
+        # @raise [UnexpectedResponse] on an unsuccessful response from the server
+        # @return [Nokogiri::XML::Document] MARCXML
+        # rubocop:disable Metrics/AbcSize
+        def marcxml(barcode: nil, catkey: nil)
+          check_args(barcode, catkey)
+
+          resp = connection.get do |req|
```

Comment:
If you cared about the ABC size of this method, you could extract this code out into a new method (response? connection? build_response? whatever), like `resp = build_response(barcode: barcode, catkey: catkey)`. I don't care that much about it. The method as written isn't difficult to understand or hold in memory.

Reply:
ABC size is now :+1: due to other changes made. Resolving.